src/test/components/TooltipMarker.test.tsx

```diff
@@ -1324,4 +1331,109 @@ describe('TooltipMarker', function () {
+            threadsKey={0}
+            restrictHeightWidth={true}
+          />
+        </Provider>
+      );
+
+      // Note: The `.` in the pattern matches bidi isolation characters that Fluent
+      // adds around variables for proper text directionality
+      const filterButton = screen.getByRole('button', {
+        name: /only show markers matching: “.Reflow.”/i,
+      });
+      expect(filterButton).toBeInTheDocument();
+    });
+
+    it('shows the filter button and uses the first field for markers with multiple fields', () => {
+      // Tooltip label: "click — DOMEvent"
+      // The filter button should use just the first field "click" as the search term
+      const profile = getProfileWithMarkers([
+        ['DOMEvent', 1, 2, { type: 'DOMEvent', eventType: 'click' }],
+      ]);
+      const store = storeWithProfile(profile);
+      const state = store.getState();
+      const getMarker = selectedThreadSelectors.getMarkerGetter(state);
+      const markerIndexes =
+        selectedThreadSelectors.getFullMarkerListIndexes(state);
+
+      render(
+        <Provider store={store}>
+          <TooltipMarker
+            markerIndex={markerIndexes[0]}
+            marker={getMarker(markerIndexes[0])}
+            threadsKey={0}
+            restrictHeightWidth={true}
+          />
+        </Provider>
+      );
+
+      // The filter button should be present and use the first field "click"
+      // Note: The `.` in the pattern matches bidi isolation characters that Fluent
+      // adds around variables for proper text directionality
+      const filterButton = screen.getByRole('button', {
+        name: /only show markers matching: “.click.”/i,
+      });
+      expect(filterButton).toBeInTheDocument();
+
+      // Verify clicking the button sets the search string to just "click"
+      fireEvent.click(filterButton);
+      expect(getMarkersSearchString(store.getState())).toBe('click');
+    });
+
+    it('sets the marker search string when the filter button is clicked', () => {
+      const profile = getProfileWithMarkers([
+        ['Reflow', 1, 2, { type: 'tracing', category: 'Paint' }],
+      ]);
+      const store = storeWithProfile(profile);
+      const state = store.getState();
+      const getMarker = selectedThreadSelectors.getMarkerGetter(state);
+      const markerIndexes =
+        selectedThreadSelectors.getFullMarkerListIndexes(state);
+
+      render(
+        <Provider store={store}>
+          <TooltipMarker
+            markerIndex={markerIndexes[0]}
+            marker={getMarker(markerIndexes[0])}
+            threadsKey={0}
+            restrictHeightWidth={true}
+          />
+        </Provider>
+      );
+
+      // Note: The `.` in the pattern matches bidi isolation characters that Fluent
+      // adds around variables for proper text directionality
+      const filterButton = screen.getByRole('button', {
+        name: /only show markers matching: “.Reflow.”/i,
```

Comment:
We usually write the isolation characters specifically, for example:
https://github.com/firefox-devtools/profiler/blob/f566a5b2eaef37144098af0bd7d16e4cb813da86/src/test/components/Timeline.test.tsx#L1258